File: 347-TopKFrequentElements/347-TopKFrequentElements.java
// Last updated: 3/14/2026, 11:50:36 AM
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int[] arr = new int[k]; 
        for(int i = 0; i< nums.length ; i++){
            map.put(nums[i] , map.getOrDefault(nums[i],0)+1);    
        }
        System.out.println(map);
        int count = 0;
        int i = 0;
        while(count < k){
            int key = -1;
            int larger = Integer.MIN_VALUE;
            for(Map.Entry<Integer,Integer> entry : map.entrySet()){
                int key1 = entry.getKey();
                int value = entry.getValue();
                if(value > larger){
                    larger = value;
                    key = key1;
                    System.out.println("k ->"+key+"v ->"+larger);
                }
            }
            arr[i++] = key;
            map.remove(key);
            count++;
        }
        return arr;
    }
}

File: 347-TopKFrequentElements/347-TopKFrequentElements.py
# Last updated: 3/14/2026, 9:12:17 PM
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = Counter(nums)
        heap = []
        
        for key, val in counter.items():
            heapq.heappush(heap, (-val, key))
        
        res = [] 
        for i in range(k):
            _, key = heapq.heappop(heap)
            res.append(key)
            
        return res
